src/halmos/utils.py

```diff
@@ -221,9 +221,6 @@ def create_solver(logic="QF_AUFBV", ctx=None, timeout=0, max_memory=0):
     # QF_AUFBV: quantifier-free bitvector + array theory: https://smtlib.cs.uiowa.edu/logics.shtml
     solver = SolverFor(logic, ctx=ctx)
 
-    # reset any remaining solver states from the default context
-    solver.reset()
-
```

Comment:
now solvers are reset at the end of their life. this differs from the previous approach, as there are two active solvers: one for setup, another for each test.